images/ceph/Makefile

```diff
@@ -131,7 +131,7 @@ list-image:
 	rm -f $(MANIFESTS_DIR)/images.txt;\
 	awk '/image:/ {print $2}' $(MANIFESTS_DIR)/operator.yaml $(MANIFESTS_DIR)/cluster.yaml | \
 	cut -d: -f2- |\
-	tee -a $(MANIFESTS_DIR)/images.txt && \
+	tee $(MANIFESTS_DIR)/images.txt && \
 	awk '/quay.io/ || /k8s.gcr.io/ {print $3}' ../../pkg/operator/ceph/csi/spec.go | \
 	cut -d= -f2- |\
-	tr -d '"' | tee -a $(MANIFESTS_DIR)/images.txt
+	tr -d '"' | tee -a $(MANIFESTS_DIR)/images.txt | uniq -u
```

Comment:
Read a bit about `uniq  -u`. It prints unique lines i.e., non-duplicate lines. So wondering how it would help in removing the duplicate entries. ?

Reply:
so I put 
``` rook/ceph:master
 quay.io/ceph/ceph:v16.2.6
 rook/ceph:master
 quay.io/ceph/ceph:v16.2.6
 quay.io/cephcsi/cephcsi:v3.4.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0
 quay.io/cephcsi/cephcsi:v3.4.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0
``` 
in `images.txt` (took sample from the runner where we are getting this error link https://github.com/rook/rook/pull/8856/checks?check_run_id=3735021728#step:5:1480)

and after running `make IMAGES=ceph build` I see 
```
 rook/ceph:master
 quay.io/ceph/ceph:v16.2.6
 quay.io/cephcsi/cephcsi:v3.4.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0
```
in images.txt. So, uniq will have uniq entries only in the file and `-u` is for right order.

Reply:
I think we need to add a `sort` to play it safe, see:

```
[leseb@tarox~] cat b
 quay.io/ceph/ceph:v16.2.6
 rook/ceph:master
 quay.io/ceph/ceph:v16.2.6
 quay.io/cephcsi/cephcsi:v3.4.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0
 quay.io/cephcsi/cephcsi:v3.4.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0

[leseb@tarox~] cat b|sort -h|uniq
 k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
 k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.3.0
 k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
 k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
 k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
 quay.io/ceph/ceph:v16.2.6
 quay.io/cephcsi/cephcsi:v3.4.0
 quay.io/csiaddons/volumereplication-operator:v0.1.0
 rook/ceph:master
```

Reply:
done